staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/envelope.go

```diff
@@ -0,0 +1,246 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package kmsv2 transforms values for storage at rest using a Envelope v2 provider
+package kmsv2
+
+import (
+	"context"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apiserver/pkg/storage/value"
+	kmstypes "k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/v2alpha1"
+	"k8s.io/apiserver/pkg/storage/value/encrypt/envelope/metrics"
+	"k8s.io/utils/lru"
+)
+
+const (
+	// KMSAPIVersion is the version of the KMS API.
+	KMSAPIVersion = "v2alpha1"
```

Comment:
so we require grpc implementors to return this value in v2 responses currently? and we'll bump this to beta and then to just v2 at GA?

Reply:
that's right! This is the same behavior as `v1beta1` today. In case of `v1` we perform this check before every RPC call to the plugin. In v2 this is only part of the health check.

Reply:
huh. since v1beta1 never got to v1, I'm having a bit of a hard time picturing how we'll simultaneously support v2beta1 and v2, but can resolve that before the beta point where we have to have some cross-release compatibility story